pallets/funding/src/tests/3_auction.rs

```diff
@@ -1,1300 +1,1699 @@
+			);
+			let plmc_existential_deposits = plmc_for_failing_bid.accounts().existential_deposits();
+			let usdt_for_bidding = MockInstantiator::calculate_auction_funding_asset_charged_with_given_price(
+				&vec![failing_bid.clone()],
+				project_metadata.minimum_price,
 			);
-		});
-	}
 
-	// Institutional bids: 0x multiplier should fail
-	let jwt = get_mock_jwt(BIDDER_2, InvestorType::Institutional, generate_did_from_account(BIDDER_2));
-	inst.execute(|| {
-		assert_noop!(
-			Pallet::<TestRuntime>::bid(
-				RuntimeOrigin::signed(BIDDER_2),
-				jwt,
-				project_id,
-				1000 * ASSET_UNIT,
-				Multiplier::force_new(0),
-				AcceptedFundingAsset::USDT
-			),
-			Error::<TestRuntime>::ForbiddenMultiplier
-		);
-	});
-	// Institutional bids: 1 - 25x multiplier should work
-	for multiplier in 1..=25u8 {
-		let jwt = get_mock_jwt(BIDDER_2, InvestorType::Institutional, generate_did_from_account(BIDDER_2));
-		let bidder_plmc = MockInstantiator::calculate_auction_plmc_charged_with_given_price(
-			&vec![(BIDDER_2, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
-			project_metadata.minimum_price,
-		);
-		let bidder_usdt = MockInstantiator::calculate_auction_funding_asset_charged_with_given_price(
-			&vec![(BIDDER_2, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
-			project_metadata.minimum_price,
-		);
-		let ed = MockInstantiator::get_ed();
-		inst.mint_plmc_to(vec![(BIDDER_2, ed).into()]);
-		inst.mint_plmc_to(bidder_plmc);
-		inst.mint_foreign_asset_to(bidder_usdt);
-		assert_ok!(inst.execute(|| Pallet::<TestRuntime>::bid(
-			RuntimeOrigin::signed(BIDDER_2),
-			jwt,
-			project_id,
-			1000 * ASSET_UNIT,
-			multiplier.try_into().unwrap(),
-			AcceptedFundingAsset::USDT
-		)));
-	}
-	// Institutional bids: >=26x multiplier should fail
-	for multiplier in 26..=50u8 {
-		let jwt = get_mock_jwt(BIDDER_2, InvestorType::Institutional, generate_did_from_account(BIDDER_2));
-		let bidder_plmc = MockInstantiator::calculate_auction_plmc_charged_with_given_price(
-			&vec![(BIDDER_2, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
-			project_metadata.minimum_price,
-		);
-		let bidder_usdt = MockInstantiator::calculate_auction_funding_asset_charged_with_given_price(
-			&vec![(BIDDER_2, 1_000 * ASSET_UNIT, Multiplier::force_new(multiplier)).into()],
-			project_metadata.minimum_price,
-		);
-		let ed = MockInstantiator::get_ed();
-		inst.mint_plmc_to(vec![(BIDDER_2, ed).into()]);
-		inst.mint_plmc_to(bidder_plmc);
-		inst.mint_foreign_asset_to(bidder_usdt);
-		inst.execute(|| {
-			assert_noop!(
-				Pallet::<TestRuntime>::bid(
+			inst.mint_plmc_to(plmc_for_failing_bid.clone());
+			inst.mint_plmc_to(plmc_existential_deposits.clone());
+			inst.mint_foreign_asset_to(usdt_for_bidding.clone());
+
+			assert_err!(inst.bid_for_users(project_id, vec![failing_bid]), Error::<TestRuntime>::TooManyBidsForUser);
+		}
+
+		#[test]
+		fn splitted_bids_cannot_exceed_user_limit_count() {
```

Comment:
Can we combine the splitted_bids_cannot_exceed_user_limit_count and cannot_bid_more_than_user_limit_count. Are nearly 99% identical, right?

Reply:
Done